webapi/src/main/scala/org/knora/webapi/messages/v2/responder/listsmessages/ListsMessagesV2.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright © 2015-2018 the contributors (see Contributors.md).
+ *
+ * This file is part of Knora.
+ *
+ * Knora is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Knora is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with Knora.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.knora.webapi.messages.v2.responder.listsmessages
+
+import org.knora.webapi._
+import org.knora.webapi.messages.admin.responder.listsmessages.ListADM
+import org.knora.webapi.messages.admin.responder.usersmessages.UserADM
+import org.knora.webapi.messages.v2.responder.{KnoraRequestV2, KnoraResponseV2}
+import org.knora.webapi.util.IriConversions._
+import org.knora.webapi.util.StringFormatter
+import org.knora.webapi.util.jsonld.{JsonLDDocument, JsonLDObject, JsonLDString}
+
+
+/**
+  * An abstract trait representing a Knora v2 API request message that can be sent to `StandoffResponderV2`.
+  */
+sealed trait ListsResponderRequestV2 extends KnoraRequestV2
+
+/**
+  * Requests a list. A successful response will be a [[ListsGetRequestV2]]
+  *
+  * @param listIris            the IRIs of the lists.
+  * @param requestingUser the user making the request.
+  */
+case class ListsGetRequestV2(listIris: Seq[IRI],
+                             requestingUser: UserADM) extends ListsResponderRequestV2
+
+/**
+  * Represents an answer to a [[ListsGetRequestV2]].
+  *
+  * @param lists the list the are to be returned.
+  */
+case class ListsGetResponseV2(lists: Vector[ListADM], preferredLangs: Option[(String, String)] = None) extends KnoraResponseV2 {
```

Comment:
@benjamingeer
 I think we should serve the list in the user's preferred language. 
I think this is very similar to what we have in `getPredicateStringLiteralObject` in `OntologyMessagesV2`. Do you think we should make a utility method for that so you could pass it a `Seq[StringLiteralV2]`?

Reply:
Yes, this is what I had in mind. It handles all possible scenarios.

What should be its parameters? Could we pass it this structure (`Map[Option[String], String]`)?

```scala
val objMap: Map[Option[String], String] = predicateInfo.objects.collect {
    case StringLiteralV2(str, maybeLang) => maybeLang -> str
}.toMap

```

Reply:
I will try this approach:

```
case class StringLiteralSequenceV2(stringLiterals: Vector[StringLiteralV2]) {

    def getPreferredLanguage(preferredLangs: Option[(String, String)]) = {

    }

}
```

In `OntologyMessagesV2` I can simply add 

```
val strLits: Seq[StringLiteralV2] = predicateInfo.objects.collect {
                    case strLit: StringLiteralV2 => strLit
                }
```

and then make an instance of the case class and call the method on it.

Reply:
Looks good to me. I would suggest not making the preferred languages an option, though. I think if you didn't have preferred languages, you wouldn't need to call the method. I would make two separate `String` parameters instead.

Reply:
Ok, thanks for the hint!

Reply:
Then I think you could add methods `getLabelInPreferredLanguage` and `getCommentInPreferredLanguage` to `ListInfoADM`, `ListNodeInfoADM`, and `ListNodeADM`, which would call `StringLiteralSequenceV2. getPreferredLanguage`.

Reply:
ok, I moved the logic in a function in https://github.com/dhlab-basel/Knora/pull/852/commits/efceb3a742b460e99a0b6f636837430fdb88877a

I will try to implement our suggestions in https://github.com/dhlab-basel/Knora/pull/852#discussion_r188195377

Reply:
Or you could just use `println(MessageUtil.toSource(list))` to regenerate source code for the whole list, and copy and paste it from the console into `SharedListsTestDataADM.scala` (which is probably how that file was made in the first place).

Reply:
> I think you could try doing that with a regex in IntelliJ.

The problem is that I had to adapt expressions that went over multiple lines, and in the end my regex was so greedy that it matched more expressions that I wanted ... At least I figured out how to use backrefs ...

In the end your hint at `MessageUtil.toSource(list))` did the job, thanks a lot!